<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <!-- Arguments -->
  <!-- Dataset to play -->
  <arg name="bag_file" default="/home/$(env USER)/data/DOALS/hauptgebaeude/sequence_1/bag.bag"/>
  <arg name="player_rate" default="1" />
  
  <!-- Drift simulation -->
  <arg name="drift_intensity" default="none" /> <!-- none, light, moderate, strong, severe -->
  <arg name="output_drifted_file_name" default="/home/$(env USER)/catkinw_ws/src/dynablox/drift_simulation/config/rollouts/doals/hauptgebaeude/sequence_1/none.csv" />
  <arg name="output_gt_file_name" default="" /> <!-- Optional, leave empty if not needed -->

  
  
  <!-- Process the data -->
  <param name="use_sim_time" value="true" />

  <!-- Play the bag -->
  <node name="player" pkg="rosbag" type="play" output="log"
   args="--quiet --rate $(arg player_rate) --clock $(arg bag_file) -d 1" required="true" /> 
   
  <!-- Drift Simulator -->
  <node name="drift_simulator" pkg="drift_simulation" type="odometry_drift_simulator" output="screen" args="--alsologtostderr">
    <remap from="pointcloud" to="/os1_cloud_node/points" />
    
  	<rosparam command="load" file="$(find drift_simulation)/config/intensities/$(arg drift_intensity).yaml" />
    <param name="global_frame_name" value="map" />
    <param name="sensor_frame_name" value="os1_lidar" />
    <param name="output_drifted_file_name" value="$(arg output_drifted_file_name)" />
    <param name="output_gt_file_name" value="$(arg output_gt_file_name)" />
  </node>
  
</launch>
